### PR TITLE
cleanup more warnings

### PR DESCRIPTION
**Pull request by adizaimi**
* cleanup more warnings
* exclude noncomdb2 directories from some warning types

### CODE REVIEW

File: mem/mem.c [hunk 1]
```diff
@@ -1129,8 +1129,10 @@ char *comdb2_strndup_static(int indx, const char *s, size_t n)
 
 struct mallinfo comdb2_mallinfo_static(int indx)
 {
+#ifdef INSECURE
     struct mallinfo empty = {0};
     STATIC_RANGE_CHECK(indx, empty);
```
Comment (riverszhang89) [inline mem/mem.c]: `STATIC_RANGE_CHECK(indx, (struct mallinfo){0});` would read neater than this.

Comment (akshatsikarwar) [reply mem/mem.c]: Should that be Unsecure?

Comment (adizaimi) [reply mem/mem.c]: That seems to work, thanks Rivers.

**Comment by akshatsikarwar** [comment]
If only gcc then test for gcc (instead of linux)

**Comment by akshatsikarwar** [comment]
`if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")`

**Comment by akshatsikarwar** [comment]
`if ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU")`

File: mem/mem.c (hunk 1 above)
Comment (akshatsikarwar) [reply mem/mem.c]: INSECURE means something else, no?

Comment (adizaimi) [reply mem/mem.c]: STATIC_RANGE_CHECK is wrapped in an ifdef INSECURE   block. I was using same, but not needed if I use Rivers' trick. Thanks.

**Pull request merged**